### ISSUE TITLE
Tabs Remix

### ISSUE THREAD

**Issue by lifeiscontent** [kentcdodds/ama#416]
Hey, would you mind giving me some quick feedback on this? I feel like it's pretty complicated, but I'm not sure how to simplify. Any pointers would be greatly appreciated! https://codesandbox.io/s/v6575jpm95

**Comment by hedgerh** [comment]
Hey, saw you hadn't received a response, so figured I'd offer some feedback.

It seems like you're exposing a lot of the implementation details to whoever is using your Tabs.  I feel like I'd probably just want something like a `TabsWithPanel` component that I could plug my tab/panel data into and have it just work.  If you want to give a developer flexibility over the tab/panel rendering, maybe you could accept render props for the tab/panel?

Here are some suggestions to clean up your current approach:

1. Could you merge `TabsState` and `Tabs`?  `Tabs` doesn't do much right now, and could just do `TabsState` job.

2. The `getItemProps`/`getPanelProps` params provided by TabsState are a little unintuitive.  I feel like TabsState should provide two things:  the id/index of the currently selected tab, and handlers to change the selected tab.  The props should be derivable inside the Tab Item/Tab Panel components from these.

3. You could maybe restructure your data like this:

```
{
  id: 'nils',
  label: 'Nils Frahm',
  panel: ( <p> Hello World </p> )
}
```
or, you could use an object for your tab data structure:

```
{
  "nils": { label: 'Nils Frahm', panel: ( <p> Hello World </p> ) },
  "foo": { label: 'Foo!', panel: ( <p> Foo Bar </p> ) }
```

4. Not sure if there are accessibility implications to this or something, but why not render only the active panel, instead of all of the panels and hiding them?  You wouldn't need your panel props, then.

5. I forgot you can pass `children` to a plain div.  That confused me in your leaf components at first glance.

Lot of cool stuff going on in your code, though.  Hope that helps!

**Comment by kentcdodds** [comment]
Hi! I'm afraid I don't have time to give a super solid answer to all my open questions right now, so I just went through and answered all my AMA questions in the form of a video: https://www.youtube.com/watch?v=y23g8L8UtoQ&list=PLV5CVI1eNcJgCrPH_e6d57KRUTiDZgs0u

A few things that folks can do to help make things better for others reading the questions and answers:

1. Link to when I start answering the question. Add `&t=xxmxxs` to the end of the URL like this: https://www.youtube.com/watch?v=y23g8L8UtoQ&list=PLV5CVI1eNcJgCrPH_e6d57KRUTiDZgs0u&t=2m5s
2. Comment with links I referenced
3. Summarize what I talk about

Thanks!
